06_timer/timer.rb

```diff
@@ -1,3 +1,24 @@
 class Timer
-  #write your code here
-end
+
+	attr_accessor :seconds
+
+	def initialize
+		@seconds = 0
+	end
+
+	def seconds=(seconds)
+		@seconds = seconds
+	end
+
+	def time_string
+		s = @seconds % 60
+		m = @seconds / 60 % 60
+		h = @seconds / 60 / 60
+		"#{padded(h)}:#{padded(m)}:#{padded(s)}"
+	end
+
+	def padded(n)
+		n < 10 ? '0'+ n.to_s : n.to_s
+	end
+
+end
```

Comment:
you don't need a setter in line 9-11 if you use `attr_accessor`, or if you want to leave those lines you can use `attr_reader` instead. Overall, you `time_string` function works fine!